driver/src/main/java/org/neo4j/driver/internal/InternalStatementResult.java

```diff
@@ -274,46 +248,34 @@ else if ( done )
     @Override
     public <T> List<T> list( Function<Record, T> mapFunction )
     {
-        if ( isEmpty() )
-        {
-            assertOpen();
-            return emptyList();
-        }
-        else if ( position == -1 && hasNext() )
+        if ( hasNext() )
```

Comment:
This is so that for the empty case we can use emptyList() which returns a constant.